XCC build failures for all intel_adsp tests/platforms

When trying to build any testcase/sample using the XCC toolchain (xt-xcc) for any of the intel_adsp boards (although i suspect this will happen for any target) there is a build failure related to `#pragma GCC diagnostic not allowed inside functions`

Please also mention any information which could help others to understand
the problem you're facing:
 - targeted any intel_adsp target and affects all testcases/samples
 - Occurs with XCC toolchain
 
**Expected behavior**
Testcase successfully build when using twister

**Impact**
Currently unable to test any intel_adsp platform with XCC

**Logs and console output**
Sample partial output for one testcase below (errors are the same for all testcases).. (same pragma related error is repeated throughout the logs):
```
INFO    -  93/388 intel_adsp_cavs25         tests/benchmarks/sched/benchmark.kernel.scheduler  FAILED Build failure (build)
-- Application: /zephyrproject/zephyr/tests/benchmarks/sched
-- Found Python3: /usr/bin/python3.8 (found suitable exact version "3.8.10") found components: Interpreter 
-- Cache files will be written to: /zephyrproject/zephyr/.cache
-- Zephyr version: 3.0.99 (/zephyrproject/zephyr)
-- Found west (found suitable version "0.12.0", minimum required is "0.7.1")
-- Board: intel_adsp_cavs25
-- Found host-tools: zephyr 0.13.2 (/opt/toolchains/zephyr-sdk-0.13.2)
-- Found dtc: /opt/toolchains/zephyr-sdk-0.13.2/sysroots/x86_64-pokysdk-linux/usr/bin/dtc (found suitable version "1.6.0", minimum required is "1.4.6")
-- Found toolchain: xcc (/opt/toolchains/xcc/install/tools)
-- Found BOARD.dts: //zephyrproject/zephyr/boards/xtensa/intel_adsp_cavs25/intel_adsp_cavs25.dts
-- Generated zephyr.dts: //zephyrproject/zephyr/twister-out/intel_adsp_cavs25/tests/benchmarks/sched/benchmark.kernel.scheduler/zephyr/zephyr.dts
-- Generated devicetree_unfixed.h: //zephyrproject/zephyr/twister-out/intel_adsp_cavs25/tests/benchmarks/sched/benchmark.kernel.scheduler/zephyr/include/generated/devicetree_unfixed.h
-- Generated device_extern.h: /zephyrproject/zephyr/twister-out/intel_adsp_cavs25/tests/benchmarks/sched/benchmark.kernel.scheduler/zephyr/include/generated/device_extern.h
-- Including generated dts.cmake file: /zephyrproject/zephyr/twister-out/intel_adsp_cavs25/tests/benchmarks/sched/benchmark.kernel.scheduler/zephyr/dts.cmake
Parsing //zephyrproject/zephyr/Kconfig
Loaded configuration '/zephyrproject/zephyr/boards/xtensa/intel_adsp_cavs25/intel_adsp_cavs25_defconfig'
Merged configuration '/zephyrproject/zephyr/tests/benchmarks/sched/prj.conf'
Configuration saved to '/zephyrproject/zephyr/twister-out/intel_adsp_cavs25/tests/benchmarks/sched/benchmark.kernel.scheduler/zephyr/.config'
Kconfig header saved to '/zephyrproject/zephyr/twister-out/intel_adsp_cavs25/tests/benchmarks/sched/benchmark.kernel.scheduler/zephyr/include/generated/autoconf.h'
-- The C compiler identification is GNU 4.2.0
-- The CXX compiler identification is GNU 4.2.0
-- The ASM compiler identification is GNU
-- Found assembler: /opt/toolchains/xcc/install/tools/RG-2017.8-linux/XtensaTools/bin/xt-xcc
-- Configuring done
-- Generating done
-- Build files have been written to: /zephyrproject/zephyr/twister-out/intel_adsp_cavs25/tests/benchmarks/sched/benchmark.kernel.scheduler
[1/134] Generating include/generated/version.h
-- Zephyr version: 3.0.99 (/zephyrproject/zephyr), build: 8c7d8ecc9aae
[2/134] Preparing syscall dependency handling

[3/134] Generating ../../../../include/generated/core-isa-dM.h
[4/134] Generating ../../../../include/generated/zsr.h
[5/134] Generating misc/generated/syscalls_subdirs.trigger
[6/134] Generating misc/generated/syscalls.json, misc/generated/struct_tags.json
[7/134] Generating include/generated/syscall_dispatch.c, include/generated/syscall_list.h
[8/134] Generating include/generated/kobj-types-enum.h, include/generated/otype-to-str.h, include/generated/otype-to-size.h
[9/134] Generating include/generated/driver-validation.h
[10/134] Building C object zephyr/CMakeFiles/offsets.dir/arch/xtensa/core/offsets/offsets.c.obj
[11/134] Generating include/generated/offsets.h
[12/134] Generating linker_zephyr_pre0.cmd
[13/134] Generating linker_zephyr_pre1.cmd
[14/134] Building ASM object zephyr/arch/arch/xtensa/core/CMakeFiles/arch__xtensa__core.dir/xtensa-asm2-util.S.obj
[15/134] Building ASM object zephyr/arch/arch/xtensa/core/CMakeFiles/arch__xtensa__core.dir/window_vectors.S.obj
[16/134] Building ASM object zephyr/arch/arch/xtensa/core/CMakeFiles/arch__xtensa__core.dir/crt1.S.obj
[17/134] Building ASM object zephyr/arch/arch/xtensa/core/startup/CMakeFiles/arch__xtensa__core__startup.dir/memerror-vector.S.obj
[18/134] Building ASM object zephyr/arch/arch/xtensa/core/startup/CMakeFiles/arch__xtensa__core__startup.dir/memctl_default.S.obj
[19/134] Building ASM object zephyr/arch/arch/xtensa/core/startup/CMakeFiles/arch__xtensa__core__startup.dir/reset-vector.S.obj
[20/134] Linking ASM static library zephyr/arch/arch/xtensa/core/startup/libarch__xtensa__core__startup.a
[21/134] Building C object zephyr/kernel/CMakeFiles/kernel.dir/fatal.c.obj
FAILED: zephyr/kernel/CMakeFiles/kernel.dir/fatal.c.obj 
/opt/toolchains/xcc/install/tools/RG-2017.8-linux/XtensaTools/bin/xt-xcc -DKERNEL -D_FORTIFY_SOURCE=2 -D__ZEPHYR_SUPERVISOR__ -D__ZEPHYR__=1 -I../../../../../../kernel/include -I../../../../../../arch/xtensa/include -I../../../../../../include -Izephyr/include/generated -I../../../../../../soc/xtensa/intel_adsp/cavs_v25 -I../../../../../../soc/xtensa/intel_adsp/common/include -I../../../../../../subsys/testsuite/include -isystem ../../../../../../lib/libc/minimal/include -Os -imacros /zephyrproject/zephyr/twister-out/intel_adsp_cavs25/tests/benchmarks/sched/benchmark.kernel.scheduler/zephyr/include/generated/autoconf.h -ffreestanding -fno-common -g -imacros/zephyrproject/zephyr/include/toolchain/xcc_missing_defs.h -fms-extensions -imacros /zephyrproject/zephyr/include/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wno-main -Wno-pointer-sign -Wpointer-arith -fgnu89-inline -Werror=implicit-int -fno-pic -ffunction-sections -fdata-sections -mlongcalls -std=c99 -MD -MT zephyr/kernel/CMakeFiles/kernel.dir/fatal.c.obj -MF zephyr/kernel/CMakeFiles/kernel.dir/fatal.c.obj.d -o zephyr/kernel/CMakeFiles/kernel.dir/fatal.c.obj -c /zephyrproject/zephyr/kernel/fatal.c
/zephyrproject/zephyr/kernel/fatal.c: In function ‘k_sys_fatal_error_handler’:
/zephyrproject/zephyr/kernel/fatal.c:43: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/kernel/fatal.c:43: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/kernel/fatal.c:43: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/kernel/fatal.c: In function ‘z_fatal_error’:
/zephyrproject/zephyr/kernel/fatal.c:108: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/kernel/fatal.c:108: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/kernel/fatal.c:107: error: ‘__auto_type’ undeclared (first use in this function)
/zephyrproject/zephyr/kernel/fatal.c:107: error: (Each undeclared identifier is reported only once
/zephyrproject/zephyr/kernel/fatal.c:107: error: for each function it appears in.)
/zephyrproject/zephyr/kernel/fatal.c:107: error: expected ‘;’ before ‘_v1’
/zephyrproject/zephyr/kernel/fatal.c:107: error: expected ‘;’ before ‘_v2’
/zephyrproject/zephyr/kernel/fatal.c:107: error: expected ‘;’ before ‘_v3’
/zephyrproject/zephyr/kernel/fatal.c:108: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/kernel/fatal.c:107: error: ‘_v1’ undeclared (first use in this function)
/zephyrproject/zephyr/kernel/fatal.c:107: error: ‘_v2’ undeclared (first use in this function)
/zephyrproject/zephyr/kernel/fatal.c:107: error: ‘_v3’ undeclared (first use in this function)
/zephyrproject/zephyr/kernel/fatal.c:123: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/kernel/fatal.c:123: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/kernel/fatal.c:122: error: expected ‘;’ before ‘_v1’
/zephyrproject/zephyr/kernel/fatal.c:122: error: expected ‘;’ before ‘_v2’
/zephyrproject/zephyr/kernel/fatal.c:123: error: #pragma GCC diagnostic not allowed inside functions
[22/134] Building C object zephyr/kernel/CMakeFiles/kernel.dir/mutex.c.obj
FAILED: zephyr/kernel/CMakeFiles/kernel.dir/mutex.c.obj 
/opt/toolchains/xcc/install/tools/RG-2017.8-linux/XtensaTools/bin/xt-xcc -DKERNEL -D_FORTIFY_SOURCE=2 -D__ZEPHYR_SUPERVISOR__ -D__ZEPHYR__=1 -I../../../../../../kernel/include -I../../../../../../arch/xtensa/include -I../../../../../../include -Izephyr/include/generated -I../../../../../../soc/xtensa/intel_adsp/cavs_v25 -I../../../../../../soc/xtensa/intel_adsp/common/include -I../../../../../../subsys/testsuite/include -isystem ../../../../../../lib/libc/minimal/include -Os -imacros /zephyrproject/zephyr/twister-out/intel_adsp_cavs25/tests/benchmarks/sched/benchmark.kernel.scheduler/zephyr/include/generated/autoconf.h -ffreestanding -fno-common -g -imacros/zephyrproject/zephyr/include/toolchain/xcc_missing_defs.h -fms-extensions -imacros /zephyrproject/zephyr/include/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wno-main -Wno-pointer-sign -Wpointer-arith -fgnu89-inline -Werror=implicit-int -fno-pic -ffunction-sections -fdata-sections -mlongcalls -std=c99 -MD -MT zephyr/kernel/CMakeFiles/kernel.dir/mutex.c.obj -MF zephyr/kernel/CMakeFiles/kernel.dir/mutex.c.obj.d -o zephyr/kernel/CMakeFiles/kernel.dir/mutex.c.obj -c /zephyrproject/zephyr/kernel/mutex.c
/zephyrproject/zephyr/kernel/mutex.c: In function ‘adjust_owner_prio’:
/zephyrproject/zephyr/kernel/mutex.c:88: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/kernel/mutex.c:88: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/kernel/mutex.c:85: error: ‘__auto_type’ undeclared (first use in this function)
/zephyrproject/zephyr/kernel/mutex.c:85: error: (Each undeclared identifier is reported only once
/zephyrproject/zephyr/kernel/mutex.c:85: error: for each function it appears in.)
/zephyrproject/zephyr/kernel/mutex.c:85: error: expected ‘;’ before ‘_v1’
/zephyrproject/zephyr/kernel/mutex.c:85: error: expected ‘;’ before ‘_v2’
/zephyrproject/zephyr/kernel/mutex.c:85: error: expected ‘;’ before ‘_v3’
/zephyrproject/zephyr/kernel/mutex.c:85: error: expected ‘;’ before ‘_v4’
/zephyrproject/zephyr/kernel/mutex.c:88: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/kernel/mutex.c:85: error: ‘_v1’ undeclared (first use in this function)
/zephyrproject/zephyr/kernel/mutex.c:85: error: ‘_v2’ undeclared (first use in this function)
/zephyrproject/zephyr/kernel/mutex.c:85: error: ‘_v3’ undeclared (first use in this function)
/zephyrproject/zephyr/kernel/mutex.c:85: error: ‘_v4’ undeclared (first use in this function)
/zephyrproject/zephyr/kernel/mutex.c: In function ‘z_impl_k_mutex_lock’:
/zephyrproject/zephyr/kernel/mutex.c:118: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/kernel/mutex.c:118: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/kernel/mutex.c:116: error: ‘__auto_type’ undeclared (first use in this function)
/zephyrproject/zephyr/kernel/mutex.c:116: error: expected ‘;’ before ‘_v1’
/zephyrproject/zephyr/kernel/mutex.c:116: error: expected ‘;’ before ‘_v2’
/zephyrproject/zephyr/kernel/mutex.c:116: error: expected ‘;’ before ‘_v3’
/zephyrproject/zephyr/kernel/mutex.c:116: error: expected ‘;’ before ‘_v4’
/zephyrproject/zephyr/kernel/mutex.c:118: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/kernel/mutex.c:116: error: ‘_v1’ undeclared (first use in this function)
/zephyrproject/zephyr/kernel/mutex.c:116: error: ‘_v2’ undeclared (first use in this function)
/zephyrproject/zephyr/kernel/mutex.c:116: error: ‘_v3’ undeclared (first use in this function)
/zephyrproject/zephyr/kernel/mutex.c:116: error: ‘_v4’ undeclared (first use in this function)
/zephyrproject/zephyr/kernel/mutex.c:140: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/kernel/mutex.c:140: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/kernel/mutex.c:140: error: expected ‘;’ before ‘_v1’
/zephyrproject/zephyr/kernel/mutex.c:140: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/kernel/mutex.c:148: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/kernel/mutex.c:148: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/kernel/mutex.c:148: error: expected ‘;’ before ‘_v1’
/zephyrproject/zephyr/kernel/mutex.c:148: error: expected ‘;’ before ‘_v2’
/zephyrproject/zephyr/kernel/mutex.c:148: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/kernel/mutex.c:151: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/kernel/mutex.c:151: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/kernel/mutex.c:150: error: expected ‘;’ before ‘_v1’
/zephyrproject/zephyr/kernel/mutex.c:150: error: expected ‘;’ before ‘_v2’
/zephyrproject/zephyr/kernel/mutex.c:150: error: expected ‘;’ before ‘_v3’
/zephyrproject/zephyr/kernel/mutex.c:151: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/kernel/mutex.c:160: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/kernel/mutex.c:160: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/kernel/mutex.c:160: error: expected ‘;’ before ‘_v1’
/zephyrproject/zephyr/kernel/mutex.c:160: error: expected ‘;’ before ‘_v2’
/zephyrproject/zephyr/kernel/mutex.c:160: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/kernel/mutex.c:170: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/kernel/mutex.c:170: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/kernel/mutex.c:170: error: expected ‘;’ before ‘_v1’
/zephyrproject/zephyr/kernel/mutex.c:170: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/kernel/mutex.c: In function ‘z_impl_k_mutex_unlock’:
/zephyrproject/zephyr/kernel/mutex.c:227: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/kernel/mutex.c:227: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/kernel/mutex.c:227: error: ‘__auto_type’ undeclared (first use in this function)
/zephyrproject/zephyr/kernel/mutex.c:227: error: expected ‘;’ before ‘_v1’
/zephyrproject/zephyr/kernel/mutex.c:227: error: expected ‘;’ before ‘_v2’
/zephyrproject/zephyr/kernel/mutex.c:227: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/kernel/mutex.c:227: error: ‘_v1’ undeclared (first use in this function)
/zephyrproject/zephyr/kernel/mutex.c:227: error: ‘_v2’ undeclared (first use in this function)
/zephyrproject/zephyr/kernel/mutex.c:248: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/kernel/mutex.c:248: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/kernel/mutex.c:247: error: expected ‘;’ before ‘_v1’
/zephyrproject/zephyr/kernel/mutex.c:247: error: expected ‘;’ before ‘_v2’
/zephyrproject/zephyr/kernel/mutex.c:247: error: expected ‘;’ before ‘_v3’
/zephyrproject/zephyr/kernel/mutex.c:248: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/kernel/mutex.c:247: error: ‘_v3’ undeclared (first use in this function)
[23/134] Building C object zephyr/arch/arch/xtensa/core/CMakeFiles/arch__xtensa__core.dir/xtensa-asm2.c.obj
FAILED: zephyr/arch/arch/xtensa/core/CMakeFiles/arch__xtensa__core.dir/xtensa-asm2.c.obj 
/opt/toolchains/xcc/install/tools/RG-2017.8-linux/XtensaTools/bin/xt-xcc -DKERNEL -D_FORTIFY_SOURCE=2 -D__ZEPHYR_SUPERVISOR__ -D__ZEPHYR__=1 -I../../../../../../kernel/include -I../../../../../../arch/xtensa/include -I../../../../../../arch/xtensa/core/include -I../../../../../../include -Izephyr/include/generated -I../../../../../../soc/xtensa/intel_adsp/cavs_v25 -I../../../../../../soc/xtensa/intel_adsp/common/include -I../../../../../../subsys/testsuite/include -isystem ../../../../../../lib/libc/minimal/include -Os -imacros /zephyrproject/zephyr/twister-out/intel_adsp_cavs25/tests/benchmarks/sched/benchmark.kernel.scheduler/zephyr/include/generated/autoconf.h -ffreestanding -fno-common -g -imacros/zephyrproject/zephyr/include/toolchain/xcc_missing_defs.h -fms-extensions -imacros /zephyrproject/zephyr/include/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wno-main -Wno-pointer-sign -Wpointer-arith -fgnu89-inline -Werror=implicit-int -fno-pic -ffunction-sections -fdata-sections -mlongcalls -std=c99 -MD -MT zephyr/arch/arch/xtensa/core/CMakeFiles/arch__xtensa__core.dir/xtensa-asm2.c.obj -MF zephyr/arch/arch/xtensa/core/CMakeFiles/arch__xtensa__core.dir/xtensa-asm2.c.obj.d -o zephyr/arch/arch/xtensa/core/CMakeFiles/arch__xtensa__core.dir/xtensa-asm2.c.obj -c /zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c: In function ‘z_irq_spurious’:
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:89: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:89: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:88: error: ‘__auto_type’ undeclared (first use in this function)
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:88: error: (Each undeclared identifier is reported only once
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:88: error: for each function it appears in.)
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:88: error: expected ‘;’ before ‘_v1’
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:88: error: expected ‘;’ before ‘_v2’
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:89: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:88: error: ‘_v1’ undeclared (first use in this function)
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:88: error: ‘_v2’ undeclared (first use in this function)
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c: In function ‘z_xtensa_dump_stack’:
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:100: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:100: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:97: error: ‘__auto_type’ undeclared (first use in this function)
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:97: error: expected ‘;’ before ‘_v1’
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:97: error: expected ‘;’ before ‘_v2’
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:97: error: expected ‘;’ before ‘_v3’
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:97: error: expected ‘;’ before ‘_v4’
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:100: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:97: error: ‘_v1’ undeclared (first use in this function)
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:97: error: ‘_v2’ undeclared (first use in this function)
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:97: error: ‘_v3’ undeclared (first use in this function)
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:97: error: ‘_v4’ undeclared (first use in this function)
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:105: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:105: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:103: error: expected ‘;’ before ‘_v1’
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:103: error: expected ‘;’ before ‘_v2’
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:103: error: expected ‘;’ before ‘_v3’
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:103: error: expected ‘;’ before ‘_v4’
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:105: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:111: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:111: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:109: error: expected ‘;’ before ‘_v1’
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:109: error: expected ‘;’ before ‘_v2’
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:109: error: expected ‘;’ before ‘_v3’
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:109: error: expected ‘;’ before ‘_v4’
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:111: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:117: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:117: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:115: error: expected ‘;’ before ‘_v1’
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:115: error: expected ‘;’ before ‘_v2’
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:115: error: expected ‘;’ before ‘_v3’
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:115: error: expected ‘;’ before ‘_v4’
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:117: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:124: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:124: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:121: error: expected ‘;’ before ‘_v1’
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:121: error: expected ‘;’ before ‘_v2’
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:121: error: expected ‘;’ before ‘_v3’
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:124: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:127: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:127: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:127: error: expected ‘;’ before ‘_v1’
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:127: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c: In function ‘xtensa_excint1_c’:
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:213: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:213: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:212: error: ‘__auto_type’ undeclared (first use in this function)
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:212: error: expected ‘;’ before ‘_v1’
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:212: error: expected ‘;’ before ‘_v2’
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:213: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:212: error: ‘_v1’ undeclared (first use in this function)
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:212: error: ‘_v2’ undeclared (first use in this function)
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:227: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:227: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:227: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:230: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:230: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:228: error: expected ‘;’ before ‘_v1’
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:228: error: expected ‘;’ before ‘_v2’
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:228: error: expected ‘;’ before ‘_v3’
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:230: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:228: error: ‘_v3’ undeclared (first use in this function)
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:232: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:232: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:231: error: expected ‘;’ before ‘_v1’
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:231: error: expected ‘;’ before ‘_v2’
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:232: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:233: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:233: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:233: error: expected ‘;’ before ‘_v1’
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:233: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:238: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:238: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:234: error: expected ‘;’ before ‘_v1’
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:234: error: expected ‘;’ before ‘_v2’
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:234: error: expected ‘;’ before ‘_v3’
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:234: error: expected ‘;’ before ‘_v4’
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:234: error: expected ‘;’ before ‘_v5’
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:234: error: expected ‘;’ before ‘_v6’
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:234: error: expected ‘;’ before ‘_v7’
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:238: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:234: error: ‘_v4’ undeclared (first use in this function)
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:234: error: ‘_v5’ undeclared (first use in this function)
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:234: error: ‘_v6’ undeclared (first use in this function)
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:234: error: ‘_v7’ undeclared (first use in this function)

```


**Environment (please complete the following information):**
 - OS: Ubuntu
 - Toolchain XCC
 - Commit SHA: f401be157f9aa7f959f18dbf371bab7db0134a5b

**Additional context**
This was initially introduced in https://github.com/zephyrproject-rtos/zephyr/pull/42165 and was reverted (which resolved the issues) now this seems to have been reintroduced within the past day and the potential fixes in https://github.com/zephyrproject-rtos/zephyr/pull/42686 do not seem to fix the errors.

Hey @nordic-krch,

So i did set `CONFIG_LOG2_ALWAYS_RUNTIME=y` and ran the full twister test suite for one target platform that was failing all the test cases and this fixed a significant amount of build failures bringing up the passing rate to 73% but there are still ~41 tests/samples that are reporting the same `#pragma GCC diagnostic not allowed inside functions` error. 

Is there another place recently introduced the pragma error that is not being caught by `CONFIG_LOG2_ALWAYS_RUNTIME=y`?

list of still failing samples and tests below:

- samples/hello_world/sample.basic.helloworld        
- samples/subsys/portability/cmsis_rtos_v2/philosophers/sample.portability.cmsis_rtos_v2.philosopher.same_prio 
- samples/philosophers/sample.kernel.philosopher.lifos 
- samples/philosophers/sample.kernel.philosopher.semaphores 
- samples/arch/smp/pktqueue/sample.smp.pktqueue      
- samples/subsys/portability/cmsis_rtos_v2/philosophers/sample.portability.cmsis_rtos_v2.philosopher.semaphores 
- samples/philosophers/sample.kernel.philosopher.same_prio 
- samples/philosophers/sample.kernel.philosopher.static 
- samples/application_development/out_of_tree_driver/sample.drivers.out_of_tree 
- samples/subsys/portability/cmsis_rtos_v2/philosophers/sample.portability.cmsis_rtos_v2.philosopher 
- samples/sensor/thermometer/sample.sensor.thermometer 
- samples/synchronization/sample.kernel.synchronization 
- samples/subsys/portability/cmsis_rtos_v1/timer_synchronization/sample.portability.cmsis_rtos_v1.timer_synchronization 
- samples/philosophers/sample.kernel.philosopher.preempt_only 
- samples/philosophers/sample.kernel.philosopher.stacks 
- samples/subsys/portability/cmsis_rtos_v1/philosophers/sample.portability.cmsis_rtos_v1.philosopher 
- samples/subsys/portability/cmsis_rtos_v1/philosophers/sample.portability.cmsis_rtos_v1.philosopher.same_prio 
- samples/subsys/tracing/tracing.user                
- samples/philosophers/sample.kernel.philosopher     
- samples/subsys/portability/cmsis_rtos_v1/philosophers/sample.portability.cmsis_rtos_v1.philosopher.semaphores 
- samples/philosophers/sample.kernel.philosopher.fifos 
- samples/philosophers/sample.kernel.philosopher.coop_only 
- samples/modules/nanopb/sample.modules.nanopb       
- samples/subsys/portability/cmsis_rtos_v2/timer_synchronization/sample.portability.cmsis_rtos_v2.timer_synchronization 
- tests/misc/test_build/buildsystem.kconfig.utf8_in_values 
- tests/posix/getopt/shell.getopt                    
- tests/subsys/shell/shell/shell.min_wildcards       
- tests/subsys/shell/shell/shell.min_tab_auto        
- tests/subsys/shell/shell/shell.min_history         
- tests/subsys/shell/shell/shell.min_metakeys        
- tests/lib/sprintf/libraries.libc.sprintf           
- tests/subsys/shell/shell/shell.min_cmds_resize     
- tests/subsys/shell/shell/shell.min_help_all        
- tests/subsys/shell/shell/shell.min_cmds_select     
- tests/subsys/shell/shell/shell.min_colors          
- tests/subsys/shell/shell/shell.min_tab             
- tests/subsys/shell/shell/shell.min_cmds_all        
- tests/subsys/shell/shell/shell.core                
- tests/subsys/shell/shell/shell.min_help            
- tests/subsys/shell/shell/shell.min                 
- tests/subsys/shell/shell/shell.min_cmds            

Full error logs for tests/subsys/shell/shell/shell.core as an example (happy to provide the full logs as well if it helps):

```
INFO    - 339/377 intel_adsp_cavs15         tests/subsys/shell/shell/shell.core                FAILED Build failure (build)
INFO    - /zephyrproject/zephyr/twister-out/intel_adsp_cavs15/tests/subsys/shell/shell/shell.core/build.log
ERROR   - Loading Zephyr default modules (Zephyr base).
-- Application: /zephyrproject/zephyr/tests/subsys/shell/shell
-- Found Python3: /usr/bin/python3.8 (found suitable exact version "3.8.10") found components: Interpreter 
-- Cache files will be written to: /zephyrproject/zephyr/.cache
-- Zephyr version: 3.0.99 (/zephyrproject/zephyr)
-- Found west (found suitable version "0.12.0", minimum required is "0.7.1")
-- Board: intel_adsp_cavs15
-- Found host-tools: zephyr 0.13.2 (/toolchains/zephyr-sdk-0.13.2)
-- Found dtc: /toolchains/zephyr-sdk-0.13.2/sysroots/x86_64-pokysdk-linux/usr/bin/dtc (found suitable version "1.6.0", minimum required is "1.4.6")
-- Found toolchain: xcc (/toolchains/xcc/install/tools)
-- Found BOARD.dts: /zephyrproject/zephyr/boards/xtensa/intel_adsp_cavs15/intel_adsp_cavs15.dts
-- Generated zephyr.dts: /zephyrproject/zephyr/twister-out/intel_adsp_cavs15/tests/subsys/shell/shell/shell.core/zephyr/zephyr.dts
-- Generated devicetree_unfixed.h: /zephyrproject/zephyr/twister-out/intel_adsp_cavs15/tests/subsys/shell/shell/shell.core/zephyr/include/generated/devicetree_unfixed.h
-- Generated device_extern.h: /zephyrproject/zephyr/twister-out/intel_adsp_cavs15/tests/subsys/shell/shell/shell.core/zephyr/include/generated/device_extern.h
-- Including generated dts.cmake file: /zephyrproject/zephyr/twister-out/intel_adsp_cavs15/tests/subsys/shell/shell/shell.core/zephyr/dts.cmake
Parsing /zephyrproject/zephyr/Kconfig
Loaded configuration '/zephyrproject/zephyr/boards/xtensa/intel_adsp_cavs15/intel_adsp_cavs15_defconfig'
Merged configuration '/zephyrproject/zephyr/tests/subsys/shell/shell/prj.conf'
Configuration saved to '/zephyrproject/zephyr/twister-out/intel_adsp_cavs15/tests/subsys/shell/shell/shell.core/zephyr/.config'
Kconfig header saved to '/zephyrproject/zephyr/twister-out/intel_adsp_cavs15/tests/subsys/shell/shell/shell.core/zephyr/include/generated/autoconf.h'
-- The C compiler identification is GNU 4.2.0
-- The CXX compiler identification is GNU 4.2.0
-- The ASM compiler identification is GNU
-- Found assembler: /toolchains/xcc/install/tools/RG-2017.8-linux/XtensaTools/bin/xt-xcc
CMake Warning at ../../../../subsys/logging/CMakeLists.txt:32 (message):
  Compiler version requires CONFIG_LOG2_ALWAYS_RUNTIME to be set


-- Configuring done
-- Generating done
-- Build files have been written to: /zephyrproject/zephyr/twister-out/intel_adsp_cavs15/tests/subsys/shell/shell/shell.core
[1/150] Preparing syscall dependency handling

[2/150] Generating include/generated/version.h
-- Zephyr version: 3.0.99 (/zephyrproject/zephyr), build: d1e45c94302b
[3/150] Generating ../../../../include/generated/core-isa-dM.h
[4/150] Generating misc/generated/syscalls_subdirs.trigger
[5/150] Generating ../../../../include/generated/zsr.h
[6/150] Generating misc/generated/syscalls.json, misc/generated/struct_tags.json
[7/150] Generating include/generated/syscall_dispatch.c, include/generated/syscall_list.h
[8/150] Generating include/generated/kobj-types-enum.h, include/generated/otype-to-str.h, include/generated/otype-to-size.h
[9/150] Generating include/generated/driver-validation.h
[10/150] Building C object zephyr/CMakeFiles/offsets.dir/arch/xtensa/core/offsets/offsets.c.obj
[11/150] Generating include/generated/offsets.h
[12/150] Generating linker_zephyr_pre0.cmd
[13/150] Generating linker_zephyr_pre1.cmd
[14/150] Building ASM object zephyr/arch/arch/xtensa/core/CMakeFiles/arch__xtensa__core.dir/window_vectors.S.obj
[15/150] Building ASM object zephyr/arch/arch/xtensa/core/CMakeFiles/arch__xtensa__core.dir/xtensa-asm2-util.S.obj
[16/150] Building ASM object zephyr/arch/arch/xtensa/core/CMakeFiles/arch__xtensa__core.dir/crt1.S.obj
[17/150] Building ASM object zephyr/arch/arch/xtensa/core/startup/CMakeFiles/arch__xtensa__core__startup.dir/memctl_default.S.obj
[18/150] Building ASM object zephyr/arch/arch/xtensa/core/startup/CMakeFiles/arch__xtensa__core__startup.dir/memerror-vector.S.obj
[19/150] Building ASM object zephyr/arch/arch/xtensa/core/startup/CMakeFiles/arch__xtensa__core__startup.dir/reset-vector.S.obj
[20/150] Linking ASM static library zephyr/arch/arch/xtensa/core/startup/libarch__xtensa__core__startup.a
[21/150] Building C object zephyr/arch/arch/xtensa/core/CMakeFiles/arch__xtensa__core.dir/fatal.c.obj
FAILED: zephyr/arch/arch/xtensa/core/CMakeFiles/arch__xtensa__core.dir/fatal.c.obj 
/toolchains/xcc/install/tools/RG-2017.8-linux/XtensaTools/bin/xt-xcc -DKERNEL -D_FORTIFY_SOURCE=2 -D__ZEPHYR_SUPERVISOR__ -D__ZEPHYR__=1 -I../../../../../../../kernel/include -I../../../../../../../arch/xtensa/include -I../../../../../../../arch/xtensa/core/include -I../../../../../../../include -Izephyr/include/generated -I../../../../../../../soc/xtensa/intel_adsp/cavs_v15 -I../../../../../../../soc/xtensa/intel_adsp/common/include -I../../../../../../../lib/util/fnmatch/. -I../../../../../../../subsys/testsuite/include -I../../../../../../../subsys/testsuite/ztest/include -isystem ../../../../../../../lib/libc/minimal/include -Os -imacros /zephyrproject/zephyr/twister-out/intel_adsp_cavs15/tests/subsys/shell/shell/shell.core/zephyr/include/generated/autoconf.h -ffreestanding -fno-common -g -imacros/zephyrproject/zephyr/include/toolchain/xcc_missing_defs.h -fms-extensions -imacros /zephyrproject/zephyr/include/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wno-main -Wno-pointer-sign -Wpointer-arith -fgnu89-inline -Werror=implicit-int -fno-pic -ffunction-sections -fdata-sections -mlongcalls -std=c99 -MD -MT zephyr/arch/arch/xtensa/core/CMakeFiles/arch__xtensa__core.dir/fatal.c.obj -MF zephyr/arch/arch/xtensa/core/CMakeFiles/arch__xtensa__core.dir/fatal.c.obj.d -o zephyr/arch/arch/xtensa/core/CMakeFiles/arch__xtensa__core.dir/fatal.c.obj -c /zephyrproject/zephyr/arch/xtensa/core/fatal.c
/zephyrproject/zephyr/arch/xtensa/core/fatal.c: In function ‘exit’:
/zephyrproject/zephyr/arch/xtensa/core/fatal.c:118: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/fatal.c:118: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/fatal.c:118: error: ‘__auto_type’ undeclared (first use in this function)
/zephyrproject/zephyr/arch/xtensa/core/fatal.c:118: error: (Each undeclared identifier is reported only once
/zephyrproject/zephyr/arch/xtensa/core/fatal.c:118: error: for each function it appears in.)
/zephyrproject/zephyr/arch/xtensa/core/fatal.c:118: error: expected ‘;’ before ‘_v1’
/zephyrproject/zephyr/arch/xtensa/core/fatal.c:118: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/fatal.c:118: error: ‘_v1’ undeclared (first use in this function)
[22/150] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/shell/modules/kernel_service.c.obj
FAILED: zephyr/CMakeFiles/zephyr.dir/subsys/shell/modules/kernel_service.c.obj 
/toolchains/xcc/install/tools/RG-2017.8-linux/XtensaTools/bin/xt-xcc -DKERNEL -D_FORTIFY_SOURCE=2 -D__ZEPHYR__=1 -I../../../../../../../kernel/include -I../../../../../../../arch/xtensa/include -I../../../../../../../include -Izephyr/include/generated -I../../../../../../../soc/xtensa/intel_adsp/cavs_v15 -I../../../../../../../soc/xtensa/intel_adsp/common/include -I../../../../../../../lib/util/fnmatch/. -I../../../../../../../subsys/testsuite/include -I../../../../../../../subsys/testsuite/ztest/include -isystem ../../../../../../../lib/libc/minimal/include -Os -imacros /zephyrproject/zephyr/twister-out/intel_adsp_cavs15/tests/subsys/shell/shell/shell.core/zephyr/include/generated/autoconf.h -ffreestanding -fno-common -g -imacros/zephyrproject/zephyr/include/toolchain/xcc_missing_defs.h -fms-extensions -imacros /zephyrproject/zephyr/include/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wno-main -Wno-pointer-sign -Wpointer-arith -fgnu89-inline -Werror=implicit-int -fno-pic -ffunction-sections -fdata-sections -mlongcalls -std=c99 -MD -MT zephyr/CMakeFiles/zephyr.dir/subsys/shell/modules/kernel_service.c.obj -MF zephyr/CMakeFiles/zephyr.dir/subsys/shell/modules/kernel_service.c.obj.d -o zephyr/CMakeFiles/zephyr.dir/subsys/shell/modules/kernel_service.c.obj -c /zephyrproject/zephyr/subsys/shell/modules/kernel_service.c
In file included from /zephyrproject/zephyr/subsys/shell/modules/kernel_service.c:13:
../../../../../../../include/debug/stack.h: In function ‘log_stack_usage’:
../../../../../../../include/debug/stack.h:36: error: #pragma GCC diagnostic not allowed inside functions
../../../../../../../include/debug/stack.h:36: error: #pragma GCC diagnostic not allowed inside functions
../../../../../../../include/debug/stack.h:34: error: ‘__auto_type’ undeclared (first use in this function)
../../../../../../../include/debug/stack.h:34: error: (Each undeclared identifier is reported only once
../../../../../../../include/debug/stack.h:34: error: for each function it appears in.)
../../../../../../../include/debug/stack.h:34: error: expected ‘;’ before ‘_v1’
../../../../../../../include/debug/stack.h:34: error: expected ‘;’ before ‘_v2’
../../../../../../../include/debug/stack.h:34: error: expected ‘;’ before ‘_v3’
../../../../../../../include/debug/stack.h:34: error: expected ‘;’ before ‘_v4’
../../../../../../../include/debug/stack.h:34: error: expected ‘;’ before ‘_v5’
../../../../../../../include/debug/stack.h:34: error: expected ‘;’ before ‘_v6’
../../../../../../../include/debug/stack.h:36: error: #pragma GCC diagnostic not allowed inside functions
../../../../../../../include/debug/stack.h:34: error: ‘_v1’ undeclared (first use in this function)
../../../../../../../include/debug/stack.h:34: error: ‘_v2’ undeclared (first use in this function)
../../../../../../../include/debug/stack.h:34: error: ‘_v3’ undeclared (first use in this function)
../../../../../../../include/debug/stack.h:34: error: ‘_v4’ undeclared (first use in this function)
../../../../../../../include/debug/stack.h:34: error: ‘_v5’ undeclared (first use in this function)
../../../../../../../include/debug/stack.h:34: error: ‘_v6’ undeclared (first use in this function)
[23/150] Building C object zephyr/lib/libc/minimal/CMakeFiles/lib__libc__minimal.dir/source/stdlib/malloc.c.obj
FAILED: zephyr/lib/libc/minimal/CMakeFiles/lib__libc__minimal.dir/source/stdlib/malloc.c.obj 
/toolchains/xcc/install/tools/RG-2017.8-linux/XtensaTools/bin/xt-xcc -DKERNEL -D_FORTIFY_SOURCE=2 -D__ZEPHYR__=1 -I../../../../../../../include -Izephyr/include/generated -I../../../../../../../soc/xtensa/intel_adsp/cavs_v15 -I../../../../../../../soc/xtensa/intel_adsp/common/include -I../../../../../../../lib/util/fnmatch/. -I../../../../../../../subsys/testsuite/include -I../../../../../../../subsys/testsuite/ztest/include -isystem ../../../../../../../lib/libc/minimal/include -Os -imacros /zephyrproject/zephyr/twister-out/intel_adsp_cavs15/tests/subsys/shell/shell/shell.core/zephyr/include/generated/autoconf.h -ffreestanding -fno-common -g -imacros/zephyrproject/zephyr/include/toolchain/xcc_missing_defs.h -fms-extensions -imacros /zephyrproject/zephyr/include/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wno-main -Wno-pointer-sign -Wpointer-arith -fgnu89-inline -Werror=implicit-int -fno-pic -ffunction-sections -fdata-sections -mlongcalls -std=c99 -MD -MT zephyr/lib/libc/minimal/CMakeFiles/lib__libc__minimal.dir/source/stdlib/malloc.c.obj -MF zephyr/lib/libc/minimal/CMakeFiles/lib__libc__minimal.dir/source/stdlib/malloc.c.obj.d -o zephyr/lib/libc/minimal/CMakeFiles/lib__libc__minimal.dir/source/stdlib/malloc.c.obj -c /zephyrproject/zephyr/lib/libc/minimal/source/stdlib/malloc.c
/zephyrproject/zephyr/lib/libc/minimal/source/stdlib/malloc.c: In function ‘malloc’:
/zephyrproject/zephyr/lib/libc/minimal/source/stdlib/malloc.c:103: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/lib/libc/minimal/source/stdlib/malloc.c:103: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/lib/libc/minimal/source/stdlib/malloc.c:103: error: #pragma GCC diagnostic not allowed inside functions
[24/150] Building C object zephyr/arch/arch/xtensa/core/CMakeFiles/arch__xtensa__core.dir/xtensa-asm2.c.obj
FAILED: zephyr/arch/arch/xtensa/core/CMakeFiles/arch__xtensa__core.dir/xtensa-asm2.c.obj 
/toolchains/xcc/install/tools/RG-2017.8-linux/XtensaTools/bin/xt-xcc -DKERNEL -D_FORTIFY_SOURCE=2 -D__ZEPHYR_SUPERVISOR__ -D__ZEPHYR__=1 -I../../../../../../../kernel/include -I../../../../../../../arch/xtensa/include -I../../../../../../../arch/xtensa/core/include -I../../../../../../../include -Izephyr/include/generated -I../../../../../../../soc/xtensa/intel_adsp/cavs_v15 -I../../../../../../../soc/xtensa/intel_adsp/common/include -I../../../../../../../lib/util/fnmatch/. -I../../../../../../../subsys/testsuite/include -I../../../../../../../subsys/testsuite/ztest/include -isystem ../../../../../../../lib/libc/minimal/include -Os -imacros /zephyrproject/zephyr/twister-out/intel_adsp_cavs15/tests/subsys/shell/shell/shell.core/zephyr/include/generated/autoconf.h -ffreestanding -fno-common -g -imacros/zephyrproject/zephyr/include/toolchain/xcc_missing_defs.h -fms-extensions -imacros /zephyrproject/zephyr/include/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wno-main -Wno-pointer-sign -Wpointer-arith -fgnu89-inline -Werror=implicit-int -fno-pic -ffunction-sections -fdata-sections -mlongcalls -std=c99 -MD -MT zephyr/arch/arch/xtensa/core/CMakeFiles/arch__xtensa__core.dir/xtensa-asm2.c.obj -MF zephyr/arch/arch/xtensa/core/CMakeFiles/arch__xtensa__core.dir/xtensa-asm2.c.obj.d -o zephyr/arch/arch/xtensa/core/CMakeFiles/arch__xtensa__core.dir/xtensa-asm2.c.obj -c /zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c: In function ‘z_irq_spurious’:
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:89: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:89: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:88: error: ‘__auto_type’ undeclared (first use in this function)
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:88: error: (Each undeclared identifier is reported only once
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:88: error: for each function it appears in.)
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:88: error: expected ‘;’ before ‘_v1’
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:88: error: expected ‘;’ before ‘_v2’
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:89: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:88: error: ‘_v1’ undeclared (first use in this function)
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:88: error: ‘_v2’ undeclared (first use in this function)
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c: In function ‘z_xtensa_dump_stack’:
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:100: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:100: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:97: error: ‘__auto_type’ undeclared (first use in this function)
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:97: error: expected ‘;’ before ‘_v1’
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:97: error: expected ‘;’ before ‘_v2’
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:97: error: expected ‘;’ before ‘_v3’
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:97: error: expected ‘;’ before ‘_v4’
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:100: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:97: error: ‘_v1’ undeclared (first use in this function)
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:97: error: ‘_v2’ undeclared (first use in this function)
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:97: error: ‘_v3’ undeclared (first use in this function)
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:97: error: ‘_v4’ undeclared (first use in this function)
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:105: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:105: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:103: error: expected ‘;’ before ‘_v1’
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:103: error: expected ‘;’ before ‘_v2’
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:103: error: expected ‘;’ before ‘_v3’
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:103: error: expected ‘;’ before ‘_v4’
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:105: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:111: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:111: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:109: error: expected ‘;’ before ‘_v1’
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:109: error: expected ‘;’ before ‘_v2’
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:109: error: expected ‘;’ before ‘_v3’
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:109: error: expected ‘;’ before ‘_v4’
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:111: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:117: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:117: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:115: error: expected ‘;’ before ‘_v1’
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:115: error: expected ‘;’ before ‘_v2’
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:115: error: expected ‘;’ before ‘_v3’
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:115: error: expected ‘;’ before ‘_v4’
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:117: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:124: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:124: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:121: error: expected ‘;’ before ‘_v1’
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:121: error: expected ‘;’ before ‘_v2’
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:121: error: expected ‘;’ before ‘_v3’
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:124: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:127: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:127: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:127: error: expected ‘;’ before ‘_v1’
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:127: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c: In function ‘xtensa_excint1_c’:
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:213: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:213: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:212: error: ‘__auto_type’ undeclared (first use in this function)
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:212: error: expected ‘;’ before ‘_v1’
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:212: error: expected ‘;’ before ‘_v2’
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:213: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:212: error: ‘_v1’ undeclared (first use in this function)
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:212: error: ‘_v2’ undeclared (first use in this function)
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:227: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:227: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:227: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:230: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:230: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:228: error: expected ‘;’ before ‘_v1’
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:228: error: expected ‘;’ before ‘_v2’
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:228: error: expected ‘;’ before ‘_v3’
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:230: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:228: error: ‘_v3’ undeclared (first use in this function)
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:232: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:232: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:231: error: expected ‘;’ before ‘_v1’
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:231: error: expected ‘;’ before ‘_v2’
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:232: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:233: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:233: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:233: error: expected ‘;’ before ‘_v1’
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:233: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:238: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:238: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:234: error: expected ‘;’ before ‘_v1’
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:234: error: expected ‘;’ before ‘_v2’
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:234: error: expected ‘;’ before ‘_v3’
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:234: error: expected ‘;’ before ‘_v4’
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:234: error: expected ‘;’ before ‘_v5’
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:234: error: expected ‘;’ before ‘_v6’
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:234: error: expected ‘;’ before ‘_v7’
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:238: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:234: error: ‘_v4’ undeclared (first use in this function)
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:234: error: ‘_v5’ undeclared (first use in this function)
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:234: error: ‘_v6’ undeclared (first use in this function)
/zephyrproject/zephyr/arch/xtensa/core/xtensa-asm2.c:234: error: ‘_v7’ undeclared (first use in this function)
[25/150] Building C object zephyr/soc/xtensa/intel_adsp/common/CMakeFiles/intel_adsp_common.dir/irq-cavs.c.obj
FAILED: zephyr/soc/xtensa/intel_adsp/common/CMakeFiles/intel_adsp_common.dir/irq-cavs.c.obj 
/toolchains/xcc/install/tools/RG-2017.8-linux/XtensaTools/bin/xt-xcc -DKERNEL -D_FORTIFY_SOURCE=2 -D__ZEPHYR__=1 -I../../../../../../../soc/xtensa/intel_adsp/common/include -I../../../../../../../drivers -I../../../../../../../include -Izephyr/include/generated -I../../../../../../../soc/xtensa/intel_adsp/cavs_v15 -I../../../../../../../lib/util/fnmatch/. -I../../../../../../../subsys/testsuite/include -I../../../../../../../subsys/testsuite/ztest/include -isystem ../../../../../../../lib/libc/minimal/include -Os -imacros /zephyrproject/zephyr/twister-out/intel_adsp_cavs15/tests/subsys/shell/shell/shell.core/zephyr/include/generated/autoconf.h -ffreestanding -fno-common -g -imacros/zephyrproject/zephyr/include/toolchain/xcc_missing_defs.h -fms-extensions -imacros /zephyrproject/zephyr/include/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wno-main -Wno-pointer-sign -Wpointer-arith -fgnu89-inline -Werror=implicit-int -fno-pic -ffunction-sections -fdata-sections -mlongcalls -std=c99 -MD -MT zephyr/soc/xtensa/intel_adsp/common/CMakeFiles/intel_adsp_common.dir/irq-cavs.c.obj -MF zephyr/soc/xtensa/intel_adsp/common/CMakeFiles/intel_adsp_common.dir/irq-cavs.c.obj.d -o zephyr/soc/xtensa/intel_adsp/common/CMakeFiles/intel_adsp_common.dir/irq-cavs.c.obj -c /zephyrproject/zephyr/soc/xtensa/intel_adsp/common/irq-cavs.c
/zephyrproject/zephyr/soc/xtensa/intel_adsp/common/irq-cavs.c: In function ‘z_soc_irq_enable’:
/zephyrproject/zephyr/soc/xtensa/intel_adsp/common/irq-cavs.c:48: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/soc/xtensa/intel_adsp/common/irq-cavs.c:48: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/soc/xtensa/intel_adsp/common/irq-cavs.c:48: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/soc/xtensa/intel_adsp/common/irq-cavs.c: In function ‘z_soc_irq_disable’:
/zephyrproject/zephyr/soc/xtensa/intel_adsp/common/irq-cavs.c:86: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/soc/xtensa/intel_adsp/common/irq-cavs.c:86: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/soc/xtensa/intel_adsp/common/irq-cavs.c:86: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/soc/xtensa/intel_adsp/common/irq-cavs.c: In function ‘z_soc_irq_is_enabled’:
/zephyrproject/zephyr/soc/xtensa/intel_adsp/common/irq-cavs.c:127: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/soc/xtensa/intel_adsp/common/irq-cavs.c:127: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/soc/xtensa/intel_adsp/common/irq-cavs.c:127: error: #pragma GCC diagnostic not allowed inside functions
[26/150] Building C object zephyr/kernel/CMakeFiles/kernel.dir/init.c.obj
FAILED: zephyr/kernel/CMakeFiles/kernel.dir/init.c.obj 
/toolchains/xcc/install/tools/RG-2017.8-linux/XtensaTools/bin/xt-xcc -DKERNEL -D_FORTIFY_SOURCE=2 -D__ZEPHYR_SUPERVISOR__ -D__ZEPHYR__=1 -I../../../../../../../kernel/include -I../../../../../../../arch/xtensa/include -I../../../../../../../include -Izephyr/include/generated -I../../../../../../../soc/xtensa/intel_adsp/cavs_v15 -I../../../../../../../soc/xtensa/intel_adsp/common/include -I../../../../../../../lib/util/fnmatch/. -I../../../../../../../subsys/testsuite/include -I../../../../../../../subsys/testsuite/ztest/include -isystem ../../../../../../../lib/libc/minimal/include -Os -imacros /zephyrproject/zephyr/twister-out/intel_adsp_cavs15/tests/subsys/shell/shell/shell.core/zephyr/include/generated/autoconf.h -ffreestanding -fno-common -g -imacros/zephyrproject/zephyr/include/toolchain/xcc_missing_defs.h -fms-extensions -imacros /zephyrproject/zephyr/include/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wno-main -Wno-pointer-sign -Wpointer-arith -fgnu89-inline -Werror=implicit-int -fno-pic -ffunction-sections -fdata-sections -mlongcalls -std=c99 -MD -MT zephyr/kernel/CMakeFiles/kernel.dir/init.c.obj -MF zephyr/kernel/CMakeFiles/kernel.dir/init.c.obj.d -o zephyr/kernel/CMakeFiles/kernel.dir/init.c.obj -c /zephyrproject/zephyr/kernel/init.c
In file included from /zephyrproject/zephyr/kernel/init.c:19:
../../../../../../../include/debug/stack.h: In function ‘log_stack_usage’:
../../../../../../../include/debug/stack.h:36: error: #pragma GCC diagnostic not allowed inside functions
../../../../../../../include/debug/stack.h:36: error: #pragma GCC diagnostic not allowed inside functions
../../../../../../../include/debug/stack.h:34: error: ‘__auto_type’ undeclared (first use in this function)
../../../../../../../include/debug/stack.h:34: error: (Each undeclared identifier is reported only once
../../../../../../../include/debug/stack.h:34: error: for each function it appears in.)
../../../../../../../include/debug/stack.h:34: error: expected ‘;’ before ‘_v1’
../../../../../../../include/debug/stack.h:34: error: expected ‘;’ before ‘_v2’
../../../../../../../include/debug/stack.h:34: error: expected ‘;’ before ‘_v3’
../../../../../../../include/debug/stack.h:34: error: expected ‘;’ before ‘_v4’
../../../../../../../include/debug/stack.h:34: error: expected ‘;’ before ‘_v5’
../../../../../../../include/debug/stack.h:34: error: expected ‘;’ before ‘_v6’
../../../../../../../include/debug/stack.h:36: error: #pragma GCC diagnostic not allowed inside functions
../../../../../../../include/debug/stack.h:34: error: ‘_v1’ undeclared (first use in this function)
../../../../../../../include/debug/stack.h:34: error: ‘_v2’ undeclared (first use in this function)
../../../../../../../include/debug/stack.h:34: error: ‘_v3’ undeclared (first use in this function)
../../../../../../../include/debug/stack.h:34: error: ‘_v4’ undeclared (first use in this function)
../../../../../../../include/debug/stack.h:34: error: ‘_v5’ undeclared (first use in this function)
../../../../../../../include/debug/stack.h:34: error: ‘_v6’ undeclared (first use in this function)
[27/150] Building C object zephyr/kernel/CMakeFiles/kernel.dir/fatal.c.obj
FAILED: zephyr/kernel/CMakeFiles/kernel.dir/fatal.c.obj 
/toolchains/xcc/install/tools/RG-2017.8-linux/XtensaTools/bin/xt-xcc -DKERNEL -D_FORTIFY_SOURCE=2 -D__ZEPHYR_SUPERVISOR__ -D__ZEPHYR__=1 -I../../../../../../../kernel/include -I../../../../../../../arch/xtensa/include -I../../../../../../../include -Izephyr/include/generated -I../../../../../../../soc/xtensa/intel_adsp/cavs_v15 -I../../../../../../../soc/xtensa/intel_adsp/common/include -I../../../../../../../lib/util/fnmatch/. -I../../../../../../../subsys/testsuite/include -I../../../../../../../subsys/testsuite/ztest/include -isystem ../../../../../../../lib/libc/minimal/include -Os -imacros /zephyrproject/zephyr/twister-out/intel_adsp_cavs15/tests/subsys/shell/shell/shell.core/zephyr/include/generated/autoconf.h -ffreestanding -fno-common -g -imacros/zephyrproject/zephyr/include/toolchain/xcc_missing_defs.h -fms-extensions -imacros /zephyrproject/zephyr/include/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wno-main -Wno-pointer-sign -Wpointer-arith -fgnu89-inline -Werror=implicit-int -fno-pic -ffunction-sections -fdata-sections -mlongcalls -std=c99 -MD -MT zephyr/kernel/CMakeFiles/kernel.dir/fatal.c.obj -MF zephyr/kernel/CMakeFiles/kernel.dir/fatal.c.obj.d -o zephyr/kernel/CMakeFiles/kernel.dir/fatal.c.obj -c /zephyrproject/zephyr/kernel/fatal.c
/zephyrproject/zephyr/kernel/fatal.c: In function ‘k_sys_fatal_error_handler’:
/zephyrproject/zephyr/kernel/fatal.c:43: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/kernel/fatal.c:43: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/kernel/fatal.c:43: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/kernel/fatal.c: In function ‘z_fatal_error’:
/zephyrproject/zephyr/kernel/fatal.c:108: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/kernel/fatal.c:108: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/kernel/fatal.c:107: error: ‘__auto_type’ undeclared (first use in this function)
/zephyrproject/zephyr/kernel/fatal.c:107: error: (Each undeclared identifier is reported only once
/zephyrproject/zephyr/kernel/fatal.c:107: error: for each function it appears in.)
/zephyrproject/zephyr/kernel/fatal.c:107: error: expected ‘;’ before ‘_v1’
/zephyrproject/zephyr/kernel/fatal.c:107: error: expected ‘;’ before ‘_v2’
/zephyrproject/zephyr/kernel/fatal.c:107: error: expected ‘;’ before ‘_v3’
/zephyrproject/zephyr/kernel/fatal.c:108: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/kernel/fatal.c:107: error: ‘_v1’ undeclared (first use in this function)
/zephyrproject/zephyr/kernel/fatal.c:107: error: ‘_v2’ undeclared (first use in this function)
/zephyrproject/zephyr/kernel/fatal.c:107: error: ‘_v3’ undeclared (first use in this function)
/zephyrproject/zephyr/kernel/fatal.c:123: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/kernel/fatal.c:123: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/kernel/fatal.c:122: error: expected ‘;’ before ‘_v1’
/zephyrproject/zephyr/kernel/fatal.c:122: error: expected ‘;’ before ‘_v2’
/zephyrproject/zephyr/kernel/fatal.c:123: error: #pragma GCC diagnostic not allowed inside functions
[28/150] Building C object zephyr/kernel/CMakeFiles/kernel.dir/thread.c.obj
FAILED: zephyr/kernel/CMakeFiles/kernel.dir/thread.c.obj 
/toolchains/xcc/install/tools/RG-2017.8-linux/XtensaTools/bin/xt-xcc -DKERNEL -D_FORTIFY_SOURCE=2 -D__ZEPHYR_SUPERVISOR__ -D__ZEPHYR__=1 -I../../../../../../../kernel/include -I../../../../../../../arch/xtensa/include -I../../../../../../../include -Izephyr/include/generated -I../../../../../../../soc/xtensa/intel_adsp/cavs_v15 -I../../../../../../../soc/xtensa/intel_adsp/common/include -I../../../../../../../lib/util/fnmatch/. -I../../../../../../../subsys/testsuite/include -I../../../../../../../subsys/testsuite/ztest/include -isystem ../../../../../../../lib/libc/minimal/include -Os -imacros /zephyrproject/zephyr/twister-out/intel_adsp_cavs15/tests/subsys/shell/shell/shell.core/zephyr/include/generated/autoconf.h -ffreestanding -fno-common -g -imacros/zephyrproject/zephyr/include/toolchain/xcc_missing_defs.h -fms-extensions -imacros /zephyrproject/zephyr/include/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wno-main -Wno-pointer-sign -Wpointer-arith -fgnu89-inline -Werror=implicit-int -fno-pic -ffunction-sections -fdata-sections -mlongcalls -std=c99 -MD -MT zephyr/kernel/CMakeFiles/kernel.dir/thread.c.obj -MF zephyr/kernel/CMakeFiles/kernel.dir/thread.c.obj.d -o zephyr/kernel/CMakeFiles/kernel.dir/thread.c.obj -c /zephyrproject/zephyr/kernel/thread.c
/zephyrproject/zephyr/kernel/thread.c: In function ‘setup_thread_stack’:
/zephyrproject/zephyr/kernel/thread.c:453: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/kernel/thread.c:453: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/kernel/thread.c:450: error: ‘__auto_type’ undeclared (first use in this function)
/zephyrproject/zephyr/kernel/thread.c:450: error: (Each undeclared identifier is reported only once
/zephyrproject/zephyr/kernel/thread.c:450: error: for each function it appears in.)
/zephyrproject/zephyr/kernel/thread.c:450: error: expected ‘;’ before ‘_v1’
/zephyrproject/zephyr/kernel/thread.c:450: error: expected ‘;’ before ‘_v2’
/zephyrproject/zephyr/kernel/thread.c:450: error: expected ‘;’ before ‘_v3’
/zephyrproject/zephyr/kernel/thread.c:450: error: expected ‘;’ before ‘_v4’
/zephyrproject/zephyr/kernel/thread.c:450: error: expected ‘;’ before ‘_v5’
/zephyrproject/zephyr/kernel/thread.c:450: error: expected ‘;’ before ‘_v6’
/zephyrproject/zephyr/kernel/thread.c:453: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/kernel/thread.c:450: error: ‘_v1’ undeclared (first use in this function)
/zephyrproject/zephyr/kernel/thread.c:450: error: ‘_v2’ undeclared (first use in this function)
/zephyrproject/zephyr/kernel/thread.c:450: error: ‘_v3’ undeclared (first use in this function)
/zephyrproject/zephyr/kernel/thread.c:450: error: ‘_v4’ undeclared (first use in this function)
/zephyrproject/zephyr/kernel/thread.c:450: error: ‘_v5’ undeclared (first use in this function)
/zephyrproject/zephyr/kernel/thread.c:450: error: ‘_v6’ undeclared (first use in this function)
[29/150] Building C object zephyr/kernel/CMakeFiles/kernel.dir/mutex.c.obj
FAILED: zephyr/kernel/CMakeFiles/kernel.dir/mutex.c.obj 
/toolchains/xcc/install/tools/RG-2017.8-linux/XtensaTools/bin/xt-xcc -DKERNEL -D_FORTIFY_SOURCE=2 -D__ZEPHYR_SUPERVISOR__ -D__ZEPHYR__=1 -I../../../../../../../kernel/include -I../../../../../../../arch/xtensa/include -I../../../../../../../include -Izephyr/include/generated -I../../../../../../../soc/xtensa/intel_adsp/cavs_v15 -I../../../../../../../soc/xtensa/intel_adsp/common/include -I../../../../../../../lib/util/fnmatch/. -I../../../../../../../subsys/testsuite/include -I../../../../../../../subsys/testsuite/ztest/include -isystem ../../../../../../../lib/libc/minimal/include -Os -imacros /zephyrproject/zephyr/twister-out/intel_adsp_cavs15/tests/subsys/shell/shell/shell.core/zephyr/include/generated/autoconf.h -ffreestanding -fno-common -g -imacros/zephyrproject/zephyr/include/toolchain/xcc_missing_defs.h -fms-extensions -imacros /zephyrproject/zephyr/include/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wno-main -Wno-pointer-sign -Wpointer-arith -fgnu89-inline -Werror=implicit-int -fno-pic -ffunction-sections -fdata-sections -mlongcalls -std=c99 -MD -MT zephyr/kernel/CMakeFiles/kernel.dir/mutex.c.obj -MF zephyr/kernel/CMakeFiles/kernel.dir/mutex.c.obj.d -o zephyr/kernel/CMakeFiles/kernel.dir/mutex.c.obj -c /zephyrproject/zephyr/kernel/mutex.c
/zephyrproject/zephyr/kernel/mutex.c: In function ‘adjust_owner_prio’:
/zephyrproject/zephyr/kernel/mutex.c:88: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/kernel/mutex.c:88: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/kernel/mutex.c:85: error: ‘__auto_type’ undeclared (first use in this function)
/zephyrproject/zephyr/kernel/mutex.c:85: error: (Each undeclared identifier is reported only once
/zephyrproject/zephyr/kernel/mutex.c:85: error: for each function it appears in.)
/zephyrproject/zephyr/kernel/mutex.c:85: error: expected ‘;’ before ‘_v1’
/zephyrproject/zephyr/kernel/mutex.c:85: error: expected ‘;’ before ‘_v2’
/zephyrproject/zephyr/kernel/mutex.c:85: error: expected ‘;’ before ‘_v3’
/zephyrproject/zephyr/kernel/mutex.c:85: error: expected ‘;’ before ‘_v4’
/zephyrproject/zephyr/kernel/mutex.c:88: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/kernel/mutex.c:85: error: ‘_v1’ undeclared (first use in this function)
/zephyrproject/zephyr/kernel/mutex.c:85: error: ‘_v2’ undeclared (first use in this function)
/zephyrproject/zephyr/kernel/mutex.c:85: error: ‘_v3’ undeclared (first use in this function)
/zephyrproject/zephyr/kernel/mutex.c:85: error: ‘_v4’ undeclared (first use in this function)
/zephyrproject/zephyr/kernel/mutex.c: In function ‘z_impl_k_mutex_lock’:
/zephyrproject/zephyr/kernel/mutex.c:118: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/kernel/mutex.c:118: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/kernel/mutex.c:116: error: ‘__auto_type’ undeclared (first use in this function)
/zephyrproject/zephyr/kernel/mutex.c:116: error: expected ‘;’ before ‘_v1’
/zephyrproject/zephyr/kernel/mutex.c:116: error: expected ‘;’ before ‘_v2’
/zephyrproject/zephyr/kernel/mutex.c:116: error: expected ‘;’ before ‘_v3’
/zephyrproject/zephyr/kernel/mutex.c:116: error: expected ‘;’ before ‘_v4’
/zephyrproject/zephyr/kernel/mutex.c:118: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/kernel/mutex.c:116: error: ‘_v1’ undeclared (first use in this function)
/zephyrproject/zephyr/kernel/mutex.c:116: error: ‘_v2’ undeclared (first use in this function)
/zephyrproject/zephyr/kernel/mutex.c:116: error: ‘_v3’ undeclared (first use in this function)
/zephyrproject/zephyr/kernel/mutex.c:116: error: ‘_v4’ undeclared (first use in this function)
/zephyrproject/zephyr/kernel/mutex.c:140: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/kernel/mutex.c:140: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/kernel/mutex.c:140: error: expected ‘;’ before ‘_v1’
/zephyrproject/zephyr/kernel/mutex.c:140: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/kernel/mutex.c:148: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/kernel/mutex.c:148: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/kernel/mutex.c:148: error: expected ‘;’ before ‘_v1’
/zephyrproject/zephyr/kernel/mutex.c:148: error: expected ‘;’ before ‘_v2’
/zephyrproject/zephyr/kernel/mutex.c:148: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/kernel/mutex.c:151: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/kernel/mutex.c:151: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/kernel/mutex.c:150: error: expected ‘;’ before ‘_v1’
/zephyrproject/zephyr/kernel/mutex.c:150: error: expected ‘;’ before ‘_v2’
/zephyrproject/zephyr/kernel/mutex.c:150: error: expected ‘;’ before ‘_v3’
/zephyrproject/zephyr/kernel/mutex.c:151: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/kernel/mutex.c:160: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/kernel/mutex.c:160: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/kernel/mutex.c:160: error: expected ‘;’ before ‘_v1’
/zephyrproject/zephyr/kernel/mutex.c:160: error: expected ‘;’ before ‘_v2’
/zephyrproject/zephyr/kernel/mutex.c:160: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/kernel/mutex.c:170: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/kernel/mutex.c:170: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/kernel/mutex.c:170: error: expected ‘;’ before ‘_v1’
/zephyrproject/zephyr/kernel/mutex.c:170: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/kernel/mutex.c: In function ‘z_impl_k_mutex_unlock’:
/zephyrproject/zephyr/kernel/mutex.c:227: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/kernel/mutex.c:227: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/kernel/mutex.c:227: error: ‘__auto_type’ undeclared (first use in this function)
/zephyrproject/zephyr/kernel/mutex.c:227: error: expected ‘;’ before ‘_v1’
/zephyrproject/zephyr/kernel/mutex.c:227: error: expected ‘;’ before ‘_v2’
/zephyrproject/zephyr/kernel/mutex.c:227: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/kernel/mutex.c:227: error: ‘_v1’ undeclared (first use in this function)
/zephyrproject/zephyr/kernel/mutex.c:227: error: ‘_v2’ undeclared (first use in this function)
/zephyrproject/zephyr/kernel/mutex.c:248: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/kernel/mutex.c:248: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/kernel/mutex.c:247: error: expected ‘;’ before ‘_v1’
/zephyrproject/zephyr/kernel/mutex.c:247: error: expected ‘;’ before ‘_v2’
/zephyrproject/zephyr/kernel/mutex.c:247: error: expected ‘;’ before ‘_v3’
/zephyrproject/zephyr/kernel/mutex.c:248: error: #pragma GCC diagnostic not allowed inside functions
/zephyrproject/zephyr/kernel/mutex.c:247: error: ‘_v3’ undeclared (first use in this function)
[30/150] Building C object zephyr/soc/xtensa/intel_adsp/common/CMakeFiles/intel_adsp_common.dir/boot.c.obj
[31/150] Building C object zephyr/kernel/CMakeFiles/kernel.dir/mem_slab.c.obj
[32/150] Building C object zephyr/soc/xtensa/intel_adsp/common/CMakeFiles/intel_adsp_common.dir/soc_mp.c.obj
[33/150] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/sem.c.obj
[34/150] Building C object zephyr/lib/posix/CMakeFiles/lib__posix.dir/nanosleep.c.obj
[35/150] Building C object zephyr/lib/libc/minimal/CMakeFiles/lib__libc__minimal.dir/source/stdlib/abort.c.obj
[36/150] Building C object zephyr/kernel/CMakeFiles/kernel.dir/sem.c.obj
[37/150] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/printk.c.obj
Warning: requested function k_spin_unlock not inlined into vprintk (edge# 359): callee is recursive
[38/150] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/assert.c.obj
[39/150] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/shell/shell_cmds.c.obj
[40/150] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/ring_buffer.c.obj
[41/150] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/shell/shell_utils.c.obj
[42/150] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/cbprintf_packaged.c.obj
[43/150] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/notify.c.obj
[44/150] Building C object zephyr/kernel/CMakeFiles/kernel.dir/version.c.obj
[45/150] Building C object zephyr/kernel/CMakeFiles/kernel.dir/mailbox.c.obj
[46/150] Building C object zephyr/kernel/CMakeFiles/kernel.dir/system_work_q.c.obj
[47/150] Building C object zephyr/arch/common/CMakeFiles/isr_tables.dir/isr_tables.c.obj
[48/150] Building C object zephyr/kernel/CMakeFiles/kernel.dir/pipes.c.obj
[49/150] Building C object zephyr/lib/libc/minimal/CMakeFiles/lib__libc__minimal.dir/source/stdlib/exit.c.obj
[50/150] Building C object zephyr/subsys/testsuite/ztest/CMakeFiles/subsys__testsuite__ztest.dir/src/ztest_error_hook.c.obj
[51/150] Building C object zephyr/soc/xtensa/intel_adsp/common/CMakeFiles/intel_adsp_common.dir/soc.c.obj
[52/150] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/crc8_sw.c.obj
[53/150] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/heap.c.obj
[54/150] Building C object zephyr/drivers/timer/CMakeFiles/drivers__timer.dir/sys_clock_init.c.obj
[55/150] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/crc7_sw.c.obj
[56/150] Building C object zephyr/kernel/CMakeFiles/kernel.dir/errno.c.obj
[57/150] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/cbprintf.c.obj
[58/150] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/fdtable.c.obj
[59/150] Building C object zephyr/kernel/CMakeFiles/kernel.dir/stack.c.obj
[60/150] Building C object zephyr/CMakeFiles/zephyr_pre0.dir/misc/empty_file.c.obj
[61/150] Building C object zephyr/drivers/interrupt_controller/CMakeFiles/drivers__interrupt_controller.dir/intc_cavs.c.obj
[62/150] Building C object zephyr/lib/libc/minimal/CMakeFiles/lib__libc__minimal.dir/source/string/strncasecmp.c.obj
[63/150] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/rb.c.obj
[64/150] Building C object zephyr/lib/libc/minimal/CMakeFiles/lib__libc__minimal.dir/source/stdlib/strtoul.c.obj
[65/150] Building C object zephyr/lib/libc/minimal/CMakeFiles/lib__libc__minimal.dir/source/stdout/stdout_console.c.obj
[66/150] Building C object zephyr/CMakeFiles/zephyr.dir/lib/util/fnmatch/fnmatch.c.obj
[67/150] Building C object zephyr/lib/libc/minimal/CMakeFiles/lib__libc__minimal.dir/source/stdlib/atoi.c.obj
[68/150] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/cbprintf_complete.c.obj
[69/150] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/winstream.c.obj
[70/150] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/crc32c_sw.c.obj
[71/150] Building C object zephyr/lib/libc/minimal/CMakeFiles/lib__libc__minimal.dir/source/stdout/fprintf.c.obj
[72/150] Building C object zephyr/kernel/CMakeFiles/kernel.dir/main_weak.c.obj
[73/150] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/shell/shell_history.c.obj
[74/150] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/onoff.c.obj
[75/150] Building C object zephyr/soc/xtensa/intel_adsp/common/CMakeFiles/intel_adsp_common.dir/rimage_modules.c.obj
[76/150] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/shell/shell_ops.c.obj
[77/150] Building C object zephyr/drivers/timer/CMakeFiles/drivers__timer.dir/cavs_timer.c.obj
[78/150] Building C object zephyr/soc/xtensa/intel_adsp/common/CMakeFiles/intel_adsp_common.dir/mp_cavs.c.obj
[79/150] Building C object zephyr/kernel/CMakeFiles/kernel.dir/device.c.obj
[80/150] Building C object zephyr/kernel/CMakeFiles/kernel.dir/kheap.c.obj
[81/150] Building C object zephyr/kernel/CMakeFiles/kernel.dir/queue.c.obj
[82/150] Building C object zephyr/arch/arch/xtensa/core/CMakeFiles/arch__xtensa__core.dir/xcc_stubs.c.obj
[83/150] Building C object zephyr/kernel/CMakeFiles/kernel.dir/banner.c.obj
[84/150] Building C object zephyr/lib/libc/minimal/CMakeFiles/lib__libc__minimal.dir/source/string/strstr.c.obj
[85/150] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/thread_entry.c.obj
[86/150] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/bitarray.c.obj
[87/150] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/crc16_sw.c.obj
[88/150] Building C object zephyr/kernel/CMakeFiles/kernel.dir/msg_q.c.obj
[89/150] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/crc32_sw.c.obj
[90/150] Building C object zephyr/kernel/CMakeFiles/kernel.dir/idle.c.obj
[91/150] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/shell/shell_fprintf.c.obj
[92/150] Building C object zephyr/arch/common/CMakeFiles/arch__common.dir/sw_isr_common.c.obj
[93/150] Building C object zephyr/arch/arch/xtensa/core/CMakeFiles/arch__xtensa__core.dir/irq_manage.c.obj
[94/150] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/dec.c.obj
[95/150] Building C object zephyr/lib/libc/minimal/CMakeFiles/lib__libc__minimal.dir/source/stdlib/qsort.c.obj
[96/150] Building C object zephyr/lib/libc/minimal/CMakeFiles/lib__libc__minimal.dir/source/stdlib/strtol.c.obj
[97/150] Building C object zephyr/lib/libc/minimal/CMakeFiles/lib__libc__minimal.dir/source/string/strspn.c.obj
[98/150] Building C object zephyr/lib/posix/CMakeFiles/lib__posix.dir/pthread_common.c.obj
[99/150] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/shell/modules/device_service.c.obj
[100/150] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/hex.c.obj
[101/150] Building C object zephyr/arch/arch/xtensa/core/CMakeFiles/arch__xtensa__core.dir/cpu_idle.c.obj
[102/150] Building C object zephyr/lib/libc/minimal/CMakeFiles/lib__libc__minimal.dir/source/stdlib/bsearch.c.obj
[103/150] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/timeutil.c.obj
[104/150] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/heap-validate.c.obj
[105/150] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/shell/shell.c.obj
[106/150] Building C object zephyr/lib/libc/minimal/CMakeFiles/lib__libc__minimal.dir/source/stdout/sprintf.c.obj
[107/150] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/shell/shell_wildcard.c.obj
[108/150] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/shell/backends/shell_dummy.c.obj
[109/150] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/shell/modules/devmem_service.c.obj
[110/150] Building C object zephyr/subsys/testsuite/ztest/CMakeFiles/subsys__testsuite__ztest.dir/src/ztest.c.obj
[111/150] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/shell/shell_help.c.obj
[112/150] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/tracing/tracing_none.c.obj
[113/150] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/multi_heap.c.obj
[114/150] Building C object zephyr/CMakeFiles/zephyr.dir/misc/generated/configs.c.obj
[115/150] Building C object zephyr/lib/libc/minimal/CMakeFiles/lib__libc__minimal.dir/source/string/string.c.obj
[116/150] Building C object zephyr/lib/libc/minimal/CMakeFiles/lib__libc__minimal.dir/source/time/gmtime.c.obj
[117/150] Building C object CMakeFiles/app.dir/src/main.c.obj
[118/150] Building C object zephyr/soc/xtensa/intel_adsp/common/CMakeFiles/intel_adsp_common.dir/trace_out.c.obj
ninja: build stopped: subcommand failed.
```

--- a/include/logging/log_msg2.h
+++ b/include/logging/log_msg2.h
@@ -12,6 +12,7 @@
 #include <sys/atomic.h>
 #include <sys/util.h>
 #include <string.h>
+#include <toolchain.h>
 
 #ifdef __GNUC__
 #ifndef alloca
@@ -323,7 +324,9 @@ do { \
  *
  * @param ...  Optional string with arguments (fmt, ...). It may be empty.
  */
-#ifdef CONFIG_LOG2_ALWAYS_RUNTIME
+#if defined(CONFIG_LOG2_ALWAYS_RUNTIME) || \
+	(!defined(CONFIG_LOG) && \
+		(!TOOLCHAIN_HAS_PRAGMA_DIAG || !TOOLCHAIN_HAS_C_AUTO_TYPE))
 #define Z_LOG_MSG2_CREATE2(_try_0cpy, _mode,  _cstr_cnt, _domain_id, _source,\
 			  _level, _data, _dlen, ...) \
 do {\
@@ -388,7 +391,9 @@ do { \
 			   _level, _data, _dlen, \
 			   FOR_EACH_IDX(Z_LOG_LOCAL_ARG_NAME, (,), __VA_ARGS__)); \
 } while (0)
-#endif /* CONFIG_LOG2_ALWAYS_RUNTIME */
+#endif /* CONFIG_LOG2_ALWAYS_RUNTIME ||
+	* (!LOG && (!TOOLCHAIN_HAS_PRAGMA_DIAG || !TOOLCHAIN_HAS_C_AUTO_TYPE))
+	*/
 
 
 #define Z_LOG_MSG2_CREATE(_try_0cpy, _mode,  _domain_id, _source,\

--- a/include/toolchain.h
+++ b/include/toolchain.h
@@ -52,6 +52,14 @@
 #error "Invalid/unknown toolchain configuration"
 #endif
 
+/**
+ * @def GCC_VERSION
+ * @brief GCC version in xxyyzz for xx.yy.zz. Zero if not GCC compatible.
+ */
+#ifndef GCC_VERSION
+#define GCC_VERSION 0
+#endif
+
 /*
  * Ensure that __BYTE_ORDER__ and related preprocessor definitions are defined,
  * as these are often used without checking for definition and doing so can

--- a/include/toolchain.h
+++ b/include/toolchain.h
@@ -92,6 +92,14 @@
 # endif
 #endif
 
+/**
+ * @def TOOLCHAIN_HAS_C_AUTO_TYPE
+ * @brief Indicate if toolchain supports C __auto_type.
+ */
+#ifndef TOOLCHAIN_HAS_C_AUTO_TYPE
+#define TOOLCHAIN_HAS_C_AUTO_TYPE 0
+#endif
+
 /*
  * Ensure that __BYTE_ORDER__ and related preprocessor definitions are defined,
  * as these are often used without checking for definition and doing so can

--- a/include/toolchain.h
+++ b/include/toolchain.h
@@ -76,6 +76,22 @@
 #define TOOLCHAIN_HAS_PRAGMA_DIAG 0
 #endif
 
+/**
+ * @def TOOLCHAIN_HAS_C_GENERIC
+ * @brief Indicate if toolchain supports C Generic.
+ */
+#if __STDC_VERSION__ >= 201112L
+/* _Generic is introduced in C11, so it is supported. */
+# ifdef TOOLCHAIN_HAS_C_GENERIC
+#  undef TOOLCHAIN_HAS_C_GENERIC
+# endif
+# define TOOLCHAIN_HAS_C_GENERIC 1
+#else
+# ifndef TOOLCHAIN_HAS_C_GENERIC
+#  define TOOLCHAIN_HAS_C_GENERIC 0
+# endif
+#endif
+
 /*
  * Ensure that __BYTE_ORDER__ and related preprocessor definitions are defined,
  * as these are often used without checking for definition and doing so can

--- a/include/toolchain.h
+++ b/include/toolchain.h
@@ -60,6 +60,14 @@
 #define GCC_VERSION 0
 #endif
 
+/**
+ * @def CLANG_VERSION
+ * @brief Clang version in xxyyzz for xx.yy.zz. Zero if not Clang compatible.
+ */
+#ifndef CLANG_VERSION
+#define CLANG_VERSION 0
+#endif
+
 /*
  * Ensure that __BYTE_ORDER__ and related preprocessor definitions are defined,
  * as these are often used without checking for definition and doing so can

--- a/include/toolchain.h
+++ b/include/toolchain.h
@@ -68,6 +68,14 @@
 #define CLANG_VERSION 0
 #endif
 
+/**
+ * @def TOOLCHAIN_HAS_PRAGMA_DIAG
+ * @brief Indicate if toolchain supports \#pragma diagnostics.
+ */
+#ifndef TOOLCHAIN_HAS_PRAGMA_DIAG
+#define TOOLCHAIN_HAS_PRAGMA_DIAG 0
+#endif
+
 /*
  * Ensure that __BYTE_ORDER__ and related preprocessor definitions are defined,
  * as these are often used without checking for definition and doing so can

--- a/include/toolchain/gcc.h
+++ b/include/toolchain/gcc.h
@@ -26,6 +26,10 @@
 #define TOOLCHAIN_HAS_C_GENERIC 1
 #endif
 
+#if !defined(TOOLCHAIN_HAS_C_AUTO_TYPE) && (GCC_VERSION >= 40900)
+#define TOOLCHAIN_HAS_C_AUTO_TYPE 1
+#endif
+
 /*
  * Older versions of GCC do not define __BYTE_ORDER__, so it must be manually
  * detected and defined using arch-specific definitions.

--- a/include/toolchain/gcc.h
+++ b/include/toolchain/gcc.h
@@ -17,6 +17,11 @@
 #define GCC_VERSION \
 	((__GNUC__ * 10000) + (__GNUC_MINOR__ * 100) + __GNUC_PATCHLEVEL__)
 
+/* GCC supports #pragma diagnostics since 4.6.0 */
+#if !defined(TOOLCHAIN_HAS_PRAGMA_DIAG) && (GCC_VERSION >= 40600)
+#define TOOLCHAIN_HAS_PRAGMA_DIAG 1
+#endif
+
 /*
  * Older versions of GCC do not define __BYTE_ORDER__, so it must be manually
  * detected and defined using arch-specific definitions.

--- a/include/toolchain/gcc.h
+++ b/include/toolchain/gcc.h
@@ -14,6 +14,9 @@
  * Macros to abstract compiler capabilities for GCC toolchain.
  */
 
+#define GCC_VERSION \
+	((__GNUC__ * 10000) + (__GNUC_MINOR__ * 100) + __GNUC_PATCHLEVEL__)
+
 /*
  * Older versions of GCC do not define __BYTE_ORDER__, so it must be manually
  * detected and defined using arch-specific definitions.

--- a/include/toolchain/gcc.h
+++ b/include/toolchain/gcc.h
@@ -22,6 +22,10 @@
 #define TOOLCHAIN_HAS_PRAGMA_DIAG 1
 #endif
 
+#if !defined(TOOLCHAIN_HAS_C_GENERIC) && (GCC_VERSION >= 40900)
+#define TOOLCHAIN_HAS_C_GENERIC 1
+#endif
+
 /*
  * Older versions of GCC do not define __BYTE_ORDER__, so it must be manually
  * detected and defined using arch-specific definitions.

--- a/include/toolchain/llvm.h
+++ b/include/toolchain/llvm.h
@@ -22,6 +22,7 @@
 
 #if CLANG_VERSION >= 30800
 #define TOOLCHAIN_HAS_C_GENERIC 1
+#define TOOLCHAIN_HAS_C_AUTO_TYPE 1
 #endif
 
 #include <toolchain/gcc.h>

--- a/include/toolchain/llvm.h
+++ b/include/toolchain/llvm.h
@@ -18,6 +18,8 @@
 	((__clang_major__ * 10000) + (__clang_minor__ * 100) + \
 	  __clang_patchlevel__)
 
+#define TOOLCHAIN_HAS_PRAGMA_DIAG 1
+
 #include <toolchain/gcc.h>
 
 

--- a/include/toolchain/llvm.h
+++ b/include/toolchain/llvm.h
@@ -14,6 +14,10 @@
 #define __fallthrough __attribute__((fallthrough))
 #endif
 
+#define CLANG_VERSION \
+	((__clang_major__ * 10000) + (__clang_minor__ * 100) + \
+	  __clang_patchlevel__)
+
 #include <toolchain/gcc.h>
 
 

--- a/include/toolchain/llvm.h
+++ b/include/toolchain/llvm.h
@@ -20,6 +20,10 @@
 
 #define TOOLCHAIN_HAS_PRAGMA_DIAG 1
 
+#if CLANG_VERSION >= 30800
+#define TOOLCHAIN_HAS_C_GENERIC 1
+#endif
+
 #include <toolchain/gcc.h>
 
 

--- a/subsys/logging/Kconfig.misc
+++ b/subsys/logging/Kconfig.misc
@@ -38,6 +38,7 @@ config LOG2_ALWAYS_RUNTIME
 	bool
 	default y if NO_OPTIMIZATIONS
 	default y if LOG_MODE_IMMEDIATE
+	default y if "$(ZEPHYR_TOOLCHAIN_VARIANT)" = "xcc"
 	help
 	  If enabled, runtime method is always used for message creation. Static
 	  creation relies on compiler being able to optimize and remove code

